### PR TITLE
ENH: add ability to locally re-map the root

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by arkilic** [comment]
@tacaswell gotta fix typo in ```filestore/test/test_fs_mutate.py::test_temporary_root```

**Comment by tacaswell** [comment]
That is not a typo, that is the test failing and it is also failing locally now.  It worked last night (I still have the console open) but fails now.

**Comment by tacaswell** [comment]
Fixed and force-pushed.

When I was removing print lines before commiting and pushing I removed the most important line from the PR :sheep:

**Comment by cowanml** [comment]
Why does it say "temporarily" everywhere?  The bulk of the usage should be permanent moves from short term to long term storage?

**Comment by tacaswell** [comment]
The use case here is caching layers or coping with a single system that has the shared storage mounted on a strange path.

This just provides a way the FS client to re-map the 'root' part of the path without changing the database.

**Comment by tacaswell** [comment]
@ericdill Can you take a look at the travis failures here?  They look conda/numpy/h5py related.

**Comment by arkilic** [comment]
Passes locally for me as well

File: filestore/fs.py [hunk 1]
```diff
@@ -71,7 +71,27 @@ def new_child(self, m=None):
 
 
 class FileStoreRO(object):
-    '''Base FileStore object that knows how to read the database.'''
+    '''Base FileStore object that knows how to read the database.
+
+    Parameters
+    ----------
+    config : dict
+       Much have keys {'database', 'collection', 'host'} and may have a 'port'
```
Comment (klauer) [inline filestore/fs.py]: 'much' -> 'must'

Comment (arkilic) [inline filestore/fs.py]: image

**Comment by ericdill** [comment]
> @ericdill Can you take a look at the travis failures here? They look conda/numpy/h5py related.

@tacaswell on it

**Comment by tacaswell** [comment]
@ericdill Thanks!

**Pull request merged**